sql/core/src/main/scala/org/apache/spark/sql/scripting/SqlScriptingExecutionNode.scala

```diff
@@ -78,11 +82,15 @@ trait NonLeafStatementExec extends CompoundStatementExec {
         case Array(field) if field.dataType == BooleanType =>
           df.limit(2).collect() match {
             case Array(row) => row.getBoolean(0)
-            case _ => false
+            case _ =>
+              throw SqlScriptingErrors.booleanStatementTooManyRows(
```

Comment:
same here, the parser only allows expression for the IF condition and we build a one row relation internally to turn the expression into a logical plan. It will never return more than one rows. We can throw internal error here.

Reply:
Updated.